code/algorithms/steprefiner.py

```diff
@@ -1,90 +1,87 @@
         """
-        rewind_steps = []
-        for vehicle_name, move in self.board.steps:
-            rewind_steps.append((vehicle_name, -move))
-        return rewind_steps
+        rewind_moves = []
+        for vehicle_name, move in self.moves:
+            rewind_moves.append((vehicle_name, -move))
+        return rewind_moves
 
 
-    def rewind_board(self, steps: int) -> None:
+    def rewind_board(self, moves: int) -> None:
         """
-        Rewind the board 'steps' number of moves.
+        Rewind the board 'moves' number of moves.
         """
-        for i in range(steps):
-            step = self.rewind_steps.pop()
-            self.board.move_vehicle(self.board.vehicles[step[0]], step[1])
+        for i in range(moves):
+            move = self.rewind_moves.pop()
+            self.mover.move_vehicle(move)
 
-    def run(self) -> Board:
+    def run(self) -> None:
         """
-        Runs the algorithm rewinding bin_size steps at a time.
+        Runs the algorithm rewinding bin_size moves at a time.
         """
         # list[list[tuple[str, int]]]
-        all_new_steps = []
-        #list[tuple[str, int]]
-        new_steps = []
+        new_moves_lists = []
 
         for bin in range(self.bins):
 
-            # save board state before further rewinding
+            # save board state before rewinding
             old_state = copy.deepcopy(self.board.locations)
             self.rewind_board(self.bin_size)
 
-            # reset steps
-            self.board.steps = []
-
             # find best path back to old_state and save it
             breadth = BreadthFirst(self.board)
             breadth.run(old_state)
-            all_new_steps.append(breadth.solution.steps)
+            new_moves_lists.append(breadth.moves)
 
-        # save board state before further rewinding
-        old_state = copy.deepcopy(self.board.locations)
-        self.rewind_board(self.last_bin_size)
+        # check for remaining moves
+        if self.last_bin_size > 0:
 
-        # reset steps
-        self.board.steps = []
-
-        # find best path back to old_state and save it
-        breadth = BreadthFirst(self.board)
-        breadth.run(old_state)
-        all_new_steps.append(breadth.solution.steps)
+            # save board state before last rewind
+            old_state = copy.deepcopy(self.board.locations)
+            self.rewind_board(self.last_bin_size)
 
-        # save steps in correct order
-        for list in reversed(all_new_steps):
-            for step in list:
-                new_steps.append(step)
+            # find best path back to old_state and save it
+            breadth = BreadthFirst(self.board)
+            breadth.run(old_state)
+            new_moves_lists.append(breadth.moves)
 
-        self.board.steps = new_steps
\ No newline at end of file
+        # save moves in correct order
+        for list in reversed(new_moves_lists):
+            for move in list:
+                self.new_moves.append(move)
```

Comment:
Think you might be able to extend the new_moves without the second for-loop

```suggestion
self.new_moves.extend(list)
```